docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,412 @@
+
+This process (steps 3-6) can also be triggered with `kubeadm phase selfhosting convert-from-staticpods`.
+
+### Running kubeadm without an internet connection
+
+All of the control plane components run in Pods started by the kubelet and
+the following images are required for the cluster works will be automatically
+pulled by the kubelet if they don't exist locally while `kubeadm init` is initializing
+your master:
+
+| Image Name |  v1.7 release branch version | v1.8 release branch version
+|---|---|---|
+| gcr.io/google_containers/kube-apiserver-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-controller-manager-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-scheduler-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/kube-proxy-${ARCH} | v1.7.x | v1.8.x
+| gcr.io/google_containers/etcd-${ARCH} | 3.0.17 | 3.0.17
+| gcr.io/google_containers/pause-${ARCH} | 3.0 | 3.0
+| gcr.io/google_containers/k8s-dns-sidecar-${ARCH} | 1.14.4 | 1.14.4
+| gcr.io/google_containers/k8s-dns-kube-dns-${ARCH} | 1.14.4 | 1.14.4
+| gcr.io/google_containers/k8s-dns-dnsmasq-nanny-${ARCH} | 1.14.4 | 1.14.4
+
+Here `v1.7.x` means the "latest patch release of the v1.7 branch".
+
+`${ARCH}` can be one of: `amd64`, `arm`, `arm64`, `ppc64le` or `s390x`.
+
+### Automating kubeadm
+
+Rather than copying the token you obtained from `kubeadm init` to each node, as
+in the [basic kubeadm tutorial](/docs/admin/kubeadm/), you can parallelize the
+token distribution for easier automation. To implement this automation, you must
+know the IP address that the master will have after it is started.
+
+1.  Generate a token. This token must have the form  `<6 character string>.<16
+    character string>`.  More formally, it must match the regex:
+    `[a-z0-9]{6}\.[a-z0-9]{16}`.
+
+    kubeadm can generate a token for you:
+
+    ```bash
+    kubeadm token generate
+    ```
+
+1. Start both the master node and the worker nodes concurrently with this token.
+   As they come up they should find each other and form the cluster.  The same
+   `--token` argument can be used on both `kubeadm init` and `kubeadm join`.
+
+Once the cluster is up, you can grab the admin credentials from the master node
+at `/etc/kubernetes/admin.conf` and use that to talk to the cluster.
+
+Note that this style of bootstrap has some relaxed security guarantees because
+it does not allow the root CA hash to be validated with
+`--discovery-token-ca-cert-hash` (since it's not generated when the nodes are
+provisioned). For details, see the [kubeadm join](kubeadm-join.md).
+
+{% endcapture %}
+
+{% capture whatsnext %}
+* [kubeadm join](kubeadm-join.md) to bootstraps a Kubernetes worker node and join it to the cluster
+* [kubeadm upgrade](kubeadm-upgrade.md) to upgrade a Kubernetes cluster to a newer version
+* [kubeadm reset](kubeadm-reset.md) to revert any changes made to this host by `kubeadm init` or `kubeadm join`
+{% endcapture %}
+
```

Comment:
how about a new section here:

```
### Using internal IPs in your cluster

In order to set up a cluster where the master and worker nodes communicate with internal IP addresses (instead of publicly addressable ones), you will need to follow these steps.

1. When running init, you must make sure you specify an internal IP for the apiserver's bind address, like so:

   kubeadm init --apiserver-advertise-address=<private-master-ip> init

2. When a worker node has been provisioned, add a flag to /etc/systemd/system/kubelet.service.d/10-kubeadm.conf that specifes the private IP of the worker node:

   --node-ip=<private-node-ip>

3. Finally, when you run `kubeadm join` make sure you provide the private IP of the apiserver addressed as defined in step 1.
```

Reply:
Done!